media_driver/agnostic/common/os/mos_gpucontextmgr.cpp

```diff
@@ -132,10 +132,31 @@ GpuContext *GpuContextMgr::CreateGpuContext(
 
     MOS_LockMutex(m_gpuContextArrayMutex);
 
-    // m_gpuContextArray always increases, directly use size before push back as search index.
-    GPU_CONTEXT_HANDLE gpuContextHandle = m_gpuContextArray.size();
+    // Directly replace nullptr with new created context in m_gpuContextArray.
+    GPU_CONTEXT_HANDLE  gpuContextHandle = 0;
+    GpuContext         *curGpuContext = nullptr;
+    int                 index = 0;
+    for (auto& curGpuContext : m_gpuContextArray)
+    {
+        if(curGpuContext == nullptr)
+        {
+            gpuContextHandle = index;
+            break;
+        }
+        index++;
+    }
+
+    gpuContextHandle = m_gpuContextArray.size() ? index : 0;
```

Comment:
Why we need reassign to gpuContextHandle here?

Reply:
I have removed reassign.